### PR TITLE
CHG: [droid] Use best possible icons for app + refactor

### PR DESCRIPTION
**Pull request by koying**
This enables hi-def icons for droid apps inside Kodi

ping @mrmc for review
/cc @MartijnKaijser

### CODE REVIEW

File: xbmc/filesystem/AndroidAppFile.h [hunk 1]
```diff
@@ -56,6 +56,7 @@ class CFileAndroidApp : public IFile
 private:
   CURL              m_url;
   std::string       m_appname;
+  androidPackage    m_droidPackage;
   int               m_iconWidth;
```
Comment (MrMC) [inline xbmc/filesystem/AndroidAppFile.h]: if you make this a ptr (androidPackage *m_droidPackage) then you can forward declare it and remove the include of android/activity/XBMCApp.h. Sure a little more work inside the cpp but filesystem includes tend to spew all over the place.

**Comment by MrMC** [comment]
other than that which is just a suggestion, seems ok, always wanted better icon support for android.

File: xbmc/filesystem/AndroidAppFile.h (hunk 1 above)
Comment (koying) [inline xbmc/filesystem/AndroidAppFile.h]: Yep. Exploding the struct would do, too, though. Just 3 fields, and 1 is a duplicate.

**Comment by MrMC** [comment]
looks ok.

**Comment by MartijnKaijser** [comment]
Tested and works ok. No significant improvement but that could be the lack of high res icons from other apps

**Comment by koying** [comment]
Beware that the app icons are probably cached, so might actually be looking at the old ones

**Comment by MartijnKaijser** [comment]
Also installed some new apps to test. Most of them also have horrible icons in play store

**Comment by koying** [comment]
Surely does make some difference:
before
after

**Pull request merged**